src/crypto/yespower-bitzeny.c

```diff
@@ -0,0 +1,50 @@
+/*-
+ * Copyright 2009 Colin Percival
+ * Copyright 2013-2018 Alexander Peslyak
+ * Copyright 2018 bitzenyPlus Developers
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions
+ * are met:
+ * 1. Redistributions of source code must retain the above copyright
+ *    notice, this list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright
+ *    notice, this list of conditions and the following disclaimer in the
+ *    documentation and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE AUTHOR AND CONTRIBUTORS ``AS IS'' AND
+ * ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE DISCLAIMED.  IN NO EVENT SHALL THE AUTHOR OR CONTRIBUTORS BE LIABLE
+ * FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+ * DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS
+ * OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION)
+ * HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ * LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY
+ * OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF
+ * SUCH DAMAGE.
+ *
+ * This file was originally written by Colin Percival as part of the Tarsnap
+ * online backup system.
+ */
+
+#include "yespower/insecure_memzero.h"
+#include "yespower/sha256.c"
+#include "yespower/sha256.h"
+#include "yespower/yespower.h"
+#include "yespower/yespower-opt.c"
```

Comment:
It's not great that you still use this hack with including everything from one file, instead of using `Makefile.am` properly. This probably works OK, but it isn't how these files were meant to be used.

Reply:
i already added it to `src/Makefile.am` but if i remove these lines from here, build is fail. i will try it again... :sob: 

https://github.com/cryptozeny/bitzeny/blob/a7b23752533d605c6744ff3cf9226ae8fcbee825/src/Makefile.am#L274

Reply:
You don't appear to understand the difference between `crypto_libbitcoin_crypto_a_SOURCES`
and `noinst_HEADERS`. In fact, it looks like some others editing this file previously didn't understand either.

Reply:
You probably only need these 3 (and no `.h` files) in `crypto_libbitcoin_crypto_a_SOURCES`:

```
  yespower/yespower-opt.c \
  yespower/sha256.c \
  crypto/yespower-bitzeny.c
```

And you probably don't need `noinst_HEADERS`, but I wouldn't vouch for that without testing.

Reply:
@solardiz 
i removed all `#include` from `/src/crypto/yespower-bitzeny.c` and added lines to `src/Makefile.am` but, it builds error...

ERR
```bash
crypto/yespower-bitzeny.c:38:55: error: ‘yespower_binary_t’ undeclared (first use in this function)
   if (yespower_tls(input, 80, &yespower_0_5_bitzeny, (yespower_binary_t *) 
                                                       ^
crypto/yespower-bitzeny.c:38:55: note: each undeclared identifier is reported only once for each function it appears in
crypto/yespower-bitzeny.c:38:74: error: expected expression before ‘)’ token
 spower_tls(input, 80, &yespower_0_5_bitzeny, (yespower_binary_t *) output))
```

:thinking:

Reply:
https://gist.github.com/cryptozeny/e690492ac36fb61357efb5a2ec353d43

Reply:
Of course, you do need the one line `#include "yespower/yespower.h"` in `yespower-bitzeny.c`.

You also mis-indented the `abort();` line. This doesn't cause any error, it's just bad style.

Reply:
You also need `#include <stdlib.h>` for `abort()` to be available.

Reply:
it seems works! thanks! :+1:

Reply:
@solardiz 

I got this error when building. gcc cannot find the file...
```bash
crypto/yespower-1.0.0/yespower-opt.c:103:31: fatal error: yespower-platform.c: No such file or directory
compilation terminated.
```

adding to `noinst_HEADERS` was a workaround. but you said it's not good.
```
noinst_HEADERS = crypto/yespower-1.0.0/yespower-platform.c
```

I see this line and is this a better fix? :thinking: 

`<crypto/yespower-1.0.0/yespower-opt.c>`
```diff
-- #include "yespower-platform.c" 
++ #include <crypto/yespower-1.0.0/yespower-platform.c>
```

Reply:
detailed error

```bash
/bin/bash ../libtool  --tag=CC   --mode=compile /home/travis/build/cryptozeny/sugarchain-v0.16.3/depends/x86_64-unknown-linux-gnu/share/../native/bin/ccache gcc -m64 -DHAVE_CONFIG_H -I. -I../src/config  -U_FORTIFY_SOURCE -D_FORTIFY_SOURCE=2 -I./obj -I./secp256k1/include -DBUILD_BITCOIN_INTERNAL -D_GLIBCXX_DEBUG -D_GLIBCXX_DEBUG_PEDANTIC -I/home/travis/build/cryptozeny/sugarchain-v0.16.3/depends/x86_64-unknown-linux-gnu/share/../include/ -DDEBUG_LOCKORDER -DHAVE_BUILD_INFO -D__STDC_FORMAT_MACROS  -pipe -O1  -c -o crypto/yespower-1.0.0/libbitcoinconsensus_la-yespower-opt.lo `test -f 'crypto/yespower-1.0.0/yespower-opt.c' || echo './'`crypto/yespower-1.0.0/yespower-opt.c
libtool: compile:  /home/travis/build/cryptozeny/sugarchain-v0.16.3/depends/x86_64-unknown-linux-gnu/share/../native/bin/ccache gcc -m64 -DHAVE_CONFIG_H -I. -I../src/config -U_FORTIFY_SOURCE -D_FORTIFY_SOURCE=2 -I./obj -I./secp256k1/include -DBUILD_BITCOIN_INTERNAL -D_GLIBCXX_DEBUG -D_GLIBCXX_DEBUG_PEDANTIC -I/home/travis/build/cryptozeny/sugarchain-v0.16.3/depends/x86_64-unknown-linux-gnu/share/../include/ -DDEBUG_LOCKORDER -DHAVE_BUILD_INFO -D__STDC_FORMAT_MACROS -pipe -O1 -c crypto/yespower-1.0.0/yespower-opt.c  -fPIC -DPIC -o crypto/yespower-1.0.0/.libs/libbitcoinconsensus_la-yespower-opt.o
crypto/yespower-1.0.0/yespower-opt.c:57:2: warning: #warning "Note: AVX and XOP are not enabled.  That's OK." [-Wcpp]
 #warning "Note: AVX and XOP are not enabled.  That's OK."
  ^
crypto/yespower-1.0.0/yespower-opt.c:103:31: fatal error: yespower-platform.c: No such file or directory
compilation terminated.
```

Reply:
I fixed it with `CFLAGS`
```bash
YESPOWER_PATH=$TRAVIS_BUILD_DIR/src/crypto/yespower-1.0.0
export CFLAGS=-I$YESPOWER_PATH
```

no errors on linux